### DISCUSSION TITLE
Help with understanding code structure

### DISCUSSION THREAD

**Discussion by letssolvesolvedproblems** [minbrowser/min#1894] [General]
I love this browser! Recently switched from Windows to Linux (Mint), and I've been slowly finding myself decreasing the complexity of the apps that I use regularly. There are a few features that I think Min could use which wouldn't clutter the experience. I've always wanted to contribute development to an open source project, but I'm a little confused about parsing the general structure of the code (something like a tree of the different source files and how they relate to each other). Could someone point me to a good resource for gaining this understanding, or offer a brief explanation? For your reference, I am not a professional software engineer, but I have some experience programming in C++, Java, and quite a bit of experience in Python (I'm a research assistant working on deep learning for medical imaging). Thank you!

**Comment by PalmerAL** [comment]
It's not entirely comprehensive, but there's a diagram here: https://github.com/minbrowser/min/wiki/Architecture. If you have specific questions beyond that I can help answer them!
